tests/filterPackages.test.ts

```diff
@@ -0,0 +1,151 @@
+
+  it("filters by minSeverity", () => {
+    const pkgs = [
+      basePkg([vuln("high")]),
+      basePkg([vuln("low")]),
+    ];
+    const filters: FilterOptions = { minSeverity: "high" };
+    const result = filterPackages(pkgs, filters);
+    expect(result.length).toBe(1);
+    expect(result[0].vulns?.[0].severity.value).toBe("high");
+  });
+
+  it("filters by packageTypes", () => {
+    const pkgs = [
+      basePkg([vuln("high")], "os"),
+      basePkg([vuln("high")], "java"),
+    ];
+    const filters: FilterOptions = { packageTypes: ["java"] };
+    const result = filterPackages(pkgs, filters);
+    expect(result.length).toBe(1);
+    expect(result[0].type).toBe("java");
+  });
+
+  it("filters by notPackageTypes", () => {
+    const pkgs = [
+      basePkg([vuln("high")], "os"),
+      basePkg([vuln("high")], "java"),
+    ];
+    const filters: FilterOptions = { notPackageTypes: ["os"] };
+    const result = filterPackages(pkgs, filters);
+    expect(result.length).toBe(1);
+    expect(result[0].type).toBe("java");
+  });
+
+  it("filters out accepted risks if excludeAccepted is true", () => {
+    const pkgs = [
+      basePkg([vuln("high", [{ index: 1, ref: "ref", id: "id" }])]),
+      basePkg([vuln("high", [])]),
+    ];
+    const filters: FilterOptions = { excludeAccepted: true };
+    const result = filterPackages(pkgs, filters);
+    expect(result.length).toBe(1);
+    expect(result[0].vulns?.[0].acceptedRisks).toEqual([]);
+  });
+
+  it("removes packages with no vulns after filtering", () => {
+    const pkgs = [
+      basePkg([vuln("low", [{ index: 1, ref: "ref", id: "id" }])]),
+    ];
+    const filters: FilterOptions = { excludeAccepted: true };
+    const result = filterPackages(pkgs, filters);
+    expect(result.length).toBe(0);
+  });
+});
+
+describe("filterPackages with fixture report", () => {
+
+  it("correctly filters the result by severity", () => {
+
+    const data: Report  = fixtureReport;
+    const filters: FilterOptions = { minSeverity: "high" };
+
+    const filteredPackages = filterPackages(data.result.packages, filters ?? {});
+    const filteredData = { ...data, result: { ...data.result, packages: filteredPackages } };
+
+    expect(JSON.stringify(filteredData)).toContain("CVE-2023-38545");
+    expect(JSON.stringify(filteredData)).not.toContain("CVE-2023-38546")
+  });
+
+  it("should return only packages with critical vulnerabilities", () => {
+    const filters: FilterOptions = { minSeverity: "Critical" as Severity };
+    const pkgs = fixtureReport.result.packages;
+    const result = filterPackages(pkgs, filters);
+
+    expect(
+      result.every(pkg => 
+        pkg.vulns?.some(v => v.severity.value.toLowerCase() === "critical")
+      )
+    ).toBe(true);
+
+    // CVE-2023-38545 is critical
+    expect(JSON.stringify(result)).toContain("CVE-2023-38545");
+    // CVE-2023-38546 is low
+    expect(JSON.stringify(result)).not.toContain("CVE-2023-38546");
+  });
```

Comment:
The first test is redundant, the second covers the same usecase and is more complete.
```suggestion
  it("should return only packages with critical vulnerabilities", () => {
    const filters: FilterOptions = { minSeverity: "Critical" as Severity };
    const pkgs = fixtureReport.result.packages;
    const result = filterPackages(pkgs, filters);

    expect(
      result.every(pkg => 
        pkg.vulns?.some(v => v.severity.value.toLowerCase() === "critical")
      )
    ).toBe(true);

    // CVE-2023-38545 is critical
    expect(JSON.stringify(result)).toContain("CVE-2023-38545");
    // CVE-2023-38546 is low
    expect(JSON.stringify(result)).not.toContain("CVE-2023-38546");
  });
```